queries.sql

```diff
@@ -1,10 +1,43 @@
 /*Queries that provide answers to the questions from all projects.*/
 
 SELECT * FROM animals WHERE name LIKE '%mon';
-SELECT name FROM animals WHERE date_of_birth >= '2016-01-01' AND date_of_birth <= '2019-12-31';
+SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
 SELECT name FROM animals WHERE neutered = true AND escape_attempts < 3;
-SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT name FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT name FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT name FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* We write queries to update and read the data in the animals table*/
+
+BEGIN;
+UPDATE animals SET species = "unspecified";
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = "digimon" WHERE name LIKE '%mon';
+UPDATE animals SET species = "pokemon" WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT kennedy;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO kennedy;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+/* We write queries to select specific conditions in the animals table*/
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT ROUND(AVG(weight_kg)::numeric, 1) FROM animals;
+SELECT name FROM animals WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```

Comment:
- [ ] Here the objective is to select which animals escape the most, nutered or unnutered animals. by using a `group by` statement.
e.g
````sql
 SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
````

Reply:
Resolved